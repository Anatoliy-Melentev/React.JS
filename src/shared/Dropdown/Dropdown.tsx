import React, { useEffect } from 'react';
import styles from './dropdown.sass';

interface IDropdownProps {
  button: React.ReactNode;
  children: React.ReactNode;
  isOpen?: boolean;
  onOpen?: () => void;
  onClose?: () => void;
}

const noop = () => {};

export function Dropdown({ button, children, isOpen, onOpen = noop, onClose = noop }: IDropdownProps) {
  const [isDropdownOpen, setIsDropdownOpen] = React.useState(isOpen);

  useEffect(() => setIsDropdownOpen(isOpen), [isOpen]);
  useEffect(() => isDropdownOpen ? onOpen() : onClose(), [isDropdownOpen]);

  const handleOpen = () => {
      if (isOpen === undefined) {
        setIsDropdownOpen(!isDropdownOpen);
      }
    };

  return (
    <div className={styles.container}>
      <div onClick={handleOpen}>
        {button}
      </div>
      {isDropdownOpen && (
        <div className={styles.listContainer}>
          <div className={styles.list} onClick={() => setIsDropdownOpen(false)}>
            {children}
          </div>
        </div>
      )}
    </div>
  );
}
