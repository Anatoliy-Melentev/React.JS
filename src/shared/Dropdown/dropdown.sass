.container
  width: 100%
  flex: 1 1 0%
  position: relative

.listContainer
  position: absolute

.list
  position: absolute
  right: 0
  left: 0
  margin: auto
  top: 20px
  width: 168px
  height: 100%
