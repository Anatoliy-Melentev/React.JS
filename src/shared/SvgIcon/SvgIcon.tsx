import React from 'react';
import styles from './svgicon.sass';
import icons from './icons.svg';

interface ISvgIcon {
  w: number;
  h: number;
  name: string;
  className?: string;
}

export function SvgIcon({ w, h, name, className = '' }: ISvgIcon) {
  return (
    <svg className={className} width={w} height={h}>
      <use xlinkHref={`${icons}#${name}`}></use>
    </svg>
  );
}
