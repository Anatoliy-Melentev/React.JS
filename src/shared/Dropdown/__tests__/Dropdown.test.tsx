/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow, mount, render } from "enzyme";
import { Dropdown } from "../Dropdown";
import styles from "../../CardsList/Card/Menu/menu.sass";
import { SvgIcon } from "../../SvgIcon";
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('Dropdown', () => {
  test('should render', () => {
    const wrapper = shallow(<Dropdown
        button={
          <button className={styles.menuButton}>
            <SvgIcon w={5} h={20} name='menu' />
          </button>
        }
        children={<div></div>}
      />);
    expect(wrapper).toBeDefined();
   // console.log(wrapper.find('div.container').debug())
    //console.log(window);
    expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy();
  })

  test('should render (shapshot)', () => {
    const wrapper = shallow(<Dropdown children={<div />} button={<button />} />);

      expect(wrapper).toMatchSnapshot()
  })
});