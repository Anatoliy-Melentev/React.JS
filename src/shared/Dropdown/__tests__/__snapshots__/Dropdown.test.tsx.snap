// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render (shapshot) 1`] = `
<div
  className="container"
>
  <div
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
